runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/util/PackageUtil.java

```diff
@@ -119,6 +122,11 @@ static PackageAttributes createPackageAttributes(File source,
       return new PackageAttributes(size, hash, directory, target, source.getPath());
     } catch (IOException e) {
       throw new RuntimeException("Package setup failure for " + source, e);
+    } finally {
+      try {
+        closer.close();
+      } catch (IOException ioe) {
```

Comment:
All the other changes look ok, my only doubt is how to deal with this case (didn't know that Close threw IOException :S).
I will probably let Dan comment on this since this is the Dataflow runner so the ball is in google's side.